src/App.tsx

```diff
@@ -1,25 +1,119 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+
+const getUserById = (userId: number): User | null => {
+  return usersFromServer.find((user) => user.id === userId) || null;
+};
+
+const todosWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [userId, setUserId] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const clearForm = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+    setHasUserIdError(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setHasTitleError(true);
+
+      return;
+    }
+
+    if (!userId) {
+      setHasUserIdError(true);
+
+      return;
+    }
+
+    const maxId = Math.max(...todos.map((todo) => todo.id)) + 1;
+
+    const createdTodo: Todo = {
+      id: maxId,
+      title,
+      userId,
+      user: getUserById(userId),
+      completed: false,
+    };
+
+    setTodos((prevTodos) => (
+      [...prevTodos, createdTodo]
+    ));
+
+    clearForm();
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          {'Title:  '}
+          <label className="labelTitle">
+            <input
+              type="text"
+              data-cy="titleInput"
+              className="labelTitle"
+              placeholder="Enter a title"
+              value={title}
+              onChange={handleTitleChange}
+            />
+          </label>
+          {hasTitleError && <span className="error">Please enter a title</span>}
```

Comment:
Maybe here is the problem with tests
```suggestion
          {hasTitleError && (<span className="error">Please enter a title</span>)}
```